src/js/utils/JSONUtil.js

```diff
@@ -224,6 +239,12 @@ module.exports = {
           } else {
             arrayIndex = -1;
           }
+
+          // Update the ending position of the last block token on stack
+          let lastBlockToken = blockTokens.pop();
```

Comment:
Curious why we are creating a stack to track the current `blockToken` (aka `lastBlockToken`) and why it's not just assigned to a variable?